### PR TITLE
Tutorial Dialogs

### PR DESCRIPTION
**Pull request by shaunanoordin**
Feature:
- When a user visits a page with a Tutorial Dialog, the Tutorial screen will pop up.
  - The tutorial can have multiple pages.
  - Each page consists of an (optional) title, an (optional) image, and either text or HTML content.
- When the tutorial dialog is closed, the '?'-help button can be clicked to open it again. 
  - After the tutorial dialog is closed for the first time, the app remembers that the user has seen that tutorial already.

screen shot 2016-08-12 at 16 25 57
_The Tutorial Dialog; note the content and paging_

screen shot 2016-08-12 at 16 26 12
_The '?'-help button_

NOTE:
- You can't actually see the Tutorial Dialog anywhere at the moment; the component is ready, but it's not inserted into any page at the moment.
- To see it in action, add to any component the following code...

```
import DialogTutorial from '../../common/components/DialogTutorial.jsx'
<DialogTutorial name="example" data={require('../../common/data/tutorial-example.js').default} />
```
- Place `<DialogTutorial/>` whereever you want to see the '?'-help button.
- `name="example"` corresponeds to the localStorage key name, so should be unique per Tutorial.

screen shot 2016-08-12 at 16 26 40

Ready for review, but will need some actual tutorial content before it makes sense to merge.

### CODE REVIEW

File: src/modules/maps/containers/MapVisuals.jsx [hunk 1]
```diff
@@ -7,8 +7,8 @@ import { MapHelper } from '../../../helpers/mapexplorer.js';
 import { enableViewCameraMode } from '../actions/mapexplorer';
 
 const mapconfig = require('../../../constants/mapExplorer.config.json');
-const gorongosaGeodata = require('../../../map-data/gorongosa-geodata.json');
-const vegetationGeodata = require('../../../map-data/vegetation-geodata.json');
+const gorongosaGeodata = require('../data/gorongosa-geodata.json');
+const vegetationGeodata = require('../data/vegetation-geodata.json');
```
Comment (simoneduca) [inline src/modules/maps/containers/MapVisuals.jsx:11]: The path should be changed here as well https://github.com/zooniverse/wildcam-gorongosa-education/blob/tutorial-dialogs/src/modules/maps/components/MapLegendVegetation.jsx#L2

Comment (shaunanoordin) [inline src/modules/maps/containers/MapVisuals.jsx:11]: Ah, I understand what you mean now! Thanks for that spot, @simoneduca, I've just fixed MapLegendVegetation. Not sure why my code compiled properly last week when it shouldn't have.

**Comment by shaunanoordin** [comment]
This PR has been updated with full Tutorial content (for Educators _and_ Explorers) and animated GIFs. Ready for a full review!

File: src/modules/common/data/tutorial-explorers.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    {
+      title: 'Welcome to the WildCam Lab!',
+      image: require('../../../images/tutorial/explorer-tutorial-1.gif'),
+      text: 'The WildCam Lab is a place for you to explore data and images from the WildCam Gorongosa project. You can use the map to explore and download data for your own project, or you can identify images and analyze your own data as part of an assignment.',
+    },
+    {
+      title: 'Explore and download data',
+      image: require('../../../images/tutorial/explorer-tutorial-2.gif'),
+      text: 'The Data page allows you to explore and download data from images that the public has identified on WildCam Gorongosa. Click the “View Filters” button and select to items to filter. Click the download button to download a spreadsheet of the data and share it with your students. Click on an orange circle to see a preview of images from that camera that match your filters.',
+    },
```
Comment (simoneduca) [inline src/modules/common/data/tutorial-explorers.js:12]: Typo: "select to items to filter" should probably be "select the items to filter".

File: src/modules/common/data/tutorial-educators.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    },
+    {
+      title: 'Create an assignment',
+      image: require('../../../images/tutorial/educator-tutorial-3.gif'),
+      text: 'An assignment is a set of WildCam images that you have selected for your students to identify and then download their own data. Within a classroom, click the assignments tab and create a new assignment. Select the number of images you want each student to identify, and select which students in your class will receive this assignment. Next, select the set of images your students will identify by filtering the map and clicking "Select for Assignment."',
+    },
+    {
+      title: 'Explore and download data',
+      image: require('../../../images/tutorial/explorer-tutorial-2.gif'),
+      text: 'The Data page allows you and your students to explore and download data from images that the public has identified on WildCam Gorongosa. Click the "View Filters" button and select to items to filter. Click the download button to download a spreadsheet of the data and share it with your students. Click on an orange circle to see a preview of images from that camera that match your filters.',
```
Comment (simoneduca) [inline src/modules/common/data/tutorial-educators.js]: Same typo: "select to items to filter" should probably be "select the items to filter".

File: src/modules/common/components/DialogTutorial.jsx [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+  }
+  
+  prevPage() {
+    const target = Math.max(parseInt(this.state.page) - 1, 0);
+    this.setState({ page: target });
+  }
+  
+  //This may be overkill - it's a mini Markdown system.
+  //Converts newlines into paragraphs, and recognises *bold* text.
+  renderPrettyText(text) {
```
Comment (simoneduca) [inline src/modules/common/components/DialogTutorial.jsx]: @shaunanoordin what's the use case for this function? You might be right it's overkill, so perhaps it's not worth bloating the code-base by including it.

File: src/styles/components/dialog-ver3.styl [hunk 5]
```diff
@@ -91,9 +99,61 @@
...
+    .paging
+      text-align: center
+      padding: 0 1em 1em 1em
+      
+      .selected
+        background: green
+        color: white
+
+.float-right
+  float: right
```
Comment (simoneduca) [inline src/styles/components/dialog-ver3.styl]: I don't think wee need this class. Use Bootstrap `pull-right` instead?

**Comment by simoneduca** [comment]
Tutorials look brilliant. Minor comments, but LGTM.

**Comment by shaunanoordin** [comment]
Thanks @simoneduca ! Your feedback has been incorporated:
- Typos fixed
- renderPrettyText() removed
- float-right replaced with Bootstrap's pull-right

**Pull request merged**